### PR TITLE
Restart stale watch threads where they left off

### PR DESCRIPTION
**Pull request by agrare**
If a watch thread fails for any reason (exception, timeout, etc...)
restart it with the last resourceVersion so it picks up where it left
off.

### CODE REVIEW

**Comment by agrare** [comment]
@cben going from your linked comment https://github.com/kubernetes/website/pull/6540#discussion_r160745657 I'm setting the collection resourceVersion to the latest notice.object's resourceVersion.  Does this look like a valid approach?

File: app/models/manageiq/providers/kubernetes/container_manager/streaming_refresh_mixin.rb [hunk 1]
```diff
@@ -103,14 +114,26 @@ def stop_watch_threads
...
 
-    until finish.value
-      watch_stream.each { |notice| queue.push(notice) }
+    until finished?
+      watch_stream.each do |notice|
+        # Update the collection resourceVersion to be the most recent
+        # object's resourceVersion so that if this watch has to be restarted
+        # it will pick up where it left off.
+        resource_version = notice.object.metadata.resourceVersion
+        resource_versions[entity_type] = resource_version
```
Comment (cben) [inline app/models/manageiq/providers/kubernetes/container_manager/streaming_refresh_mixin.rb]: this updates a shared hash (at distinct keys) from multiple threads, is that fully safe in ruby?

**Comment by cben** [comment]
Merging.  I'll be on vacation all next week so will try to unblock as much stuff as possible today.

File: app/models/manageiq/providers/kubernetes/container_manager/streaming_refresh_mixin.rb (hunk 1 above)
Comment (agrare) [reply app/models/manageiq/providers/kubernetes/container_manager/streaming_refresh_mixin.rb]: Yeah I'll use a `Concurrent::Hash` to be safe

**Pull request merged**